### ISSUE TITLE
Bitfield serialization is very go specific

### ISSUE THREAD

**Issue by dignifiedquire** [filecoin-project/go-hamt-ipld#27]
It seems that the current implementation of serializing the bitfield is hard to follow, as it depends on details of the go `BigInt` details

When printing out the results of https://github.com/ipfs/go-hamt-ipld/blob/master/hamt_test.go#L441 (`cs.Get(tcid)`) with cbor.me, I get the following.

This indicates the `Bitfield` is stored (key: "bf" at the end) as 30 bytes. 
To make it easy to align with other implementations I believe this should be normalized to one of the following

- bytes: properly padded to 8 bytes, e.g 4*8 bytes in the current config
- 1: cbor bigint type
- 4: u64 cbor type

```
A2                                      # map(2)
   61                                   # text(1)
      70                                # "p"
   81                                   # array(1)
      A1                                # map(1)
         61                             # text(1)
            76                          # "v"
         81                             # array(1)
            82                          # array(2)
               63                       # text(3)
                  636174                # "cat"
               A2                       # map(2)
                  63                    # text(3)
                     666F6F             # "foo"
                  63                    # text(3)
                     626172             # "bar"
                  63                    # text(3)
                     6F6E65             # "one"
                  D8 2A                 # tag(42)
                     58 25              # bytes(37)
                        00017112204C287CF5C5E6F4B60DB56B283B3EA8DBCBEB77F75D6EB7D654A6D38D2FEC6118 # "\x00\x01q\x12 L(|\xF5\xC5\xE6\xF4\xB6\r\xB5k(;>\xA8\xDB\xCB\xEBw\xF7]n\xB7\xD6T\xA6\xD3\x8D/\xECa\x18"
   62                                   # text(2)
      6266                              # "bf"
   58 1E                                # bytes(30)
      020000000000000000000000000000000000000000000000000000000000 # "\x02\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"

```

**Comment by dignifiedquire** [comment]
The reason I believe `4*8 = 32 bytes` is the right storage size, is because the values that are used to index into the bitmap are in the range of an `8bit` integer, meaning they range from `0-255`, meaning `256` bits to address, divided by `8`, this gives me `32 bytes`.

**Comment by austinabell** [comment]
This was linked but I think should be closed now, the serialization has since changed to RLE byte string

**Comment by Stebalien** [comment]
The hamt node bitfield. The bitfield is still a go big.Int.

**Comment by ZenGround0** [comment]
Since go-hamt-ipld has moved to the filecoin project and has narrowed in scope to become the filecoin hamt this issue is not currently relevant.  The filecoin chain data model relies on go specific big int serialization throughout the state tree and block header graphs in non-hamt data.  If we do want to modify the go language bitfield coupling for filecoin chain data we should do it at the system level.  If we make that decision down the road we should then reopen this issue with the decision to make a system-wide change as motivation.

fyi @rvagg who has helped me understand some of this context.
